### ISSUE TITLE
Build warnings in latest VS

### ISSUE THREAD

**Issue by ErikEJ** [rr-wfm/MSBuild.Sdk.SqlProj#462]
```
1>C:\Users\ErikEjlskovJensen(De\.nuget\packages\msbuild.sdk.sqlproj\2.6.1\Sdk\Sdk.targets(117,9): message : MSB4120: Item '_ResolvedPackageReference' definition within target references itself via (qualified or unqualified) metadatum 'PackageName'. This can lead to unintended expansion and cross-applying of pre-existing items. More info: https://aka.ms/msbuild/metadata-self-ref
1>C:\Users\ErikEjlskovJensen(De\.nuget\packages\msbuild.sdk.sqlproj\2.6.1\Sdk\Sdk.targets(124,9): message : MSB4120: Item '_ResolvedPackageReference' definition within target references itself via (qualified or unqualified) metadatum 'PhysicalLocation'. This can lead to unintended expansion and cross-applying of pre-existing items. More info: https://aka.ms/msbuild/metadata-self-ref

```

Are they something that should be considered @jmezach  ?

**Comment by jmezach** [comment]
@ErikEJ I think I've noticed that too, but haven't looked into it much yet. I guess they've changed something in newer versions of MSBuild. So far I don't think it is an issue, but we might have to fix it at some point.

**Comment by cagrin** [comment]
When I use global.json to setup dotnet version, this problem does not appears at dotnet 6.0.414. It show up only at dotnet 7.0.401 and 8.0.100-rc.1-23455.8. Perhaps this happens because something has been added relatively recently at versions 7 and 8 respectively?

**Comment by ErikEJ** [comment]
Looks like this is fixed with VS 17.8 and .NET 8.0 RTM
